.aegir.ts

```diff
@@ -0,0 +1,7 @@
+const options: import('./src/types').PartialOptions = {
```

Comment:
That's interesting, I didn't know you could do inline imports of types like this.

That said, is it better than `import type { Foo } from './bar.js'`?

Also, I see there's no extension on `./src/types` - does this mean this file gets compiled to CJS in the background?  What happens if you try to import an ESM-only module in your `.aegir` file?

Reply:
let me look into that

Reply:
> That said, is it better than import type { Foo } from './bar.js'?

Nope! definitely not. fixing this.

> Also, I see there's no extension on ./src/types - does this mean this file gets compiled to CJS in the background? 

Not sure, but fixing this.

> What happens if you try to import an ESM-only module in your .aegir file?

updated to 
```
import type { PartialOptions } from './src/types.js'
import { isWritableStream } from 'is-stream'
console.log('isWritableStream: ', isWritableStream)

const options: PartialOptions = {
  docs: {
    entryPoint: 'utils'
  }
}

export default options
```

and ran `npm run lint`:

```bash
npm run lint

> aegir@37.3.0 lint
> node src/index.js lint

Unexpected error while loading your config file
Unexpected error loading aegir config
/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/cache/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/.aegir.js:3
var is_stream_1 = require("is-stream");
                  ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/is-stream/index.js from /Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/cache/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/.aegir.js not supported.
Instead change the require of index.js in /Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/cache/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/.aegir.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/cache/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/.aegir.js:3:19)
    at /Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/dist/main.js:34:55
    at async Object.load (/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/ts-import/dist/main.js:34:20)
    at async loadTs (file:///Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/src/config/user.js:61:11)
    at async Object.search (/Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/node_modules/lilconfig/dist/index.js:126:37)
    at async config (file:///Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/src/config/user.js:110:26)
    at async main (file:///Users/sgtpooki/code/work/protocol.ai/ipfs/aegir/src/index.js:48:22) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v17.4.0
```

Reply:
Yeah, this is what I thought might happen.  It might be worth opening an issue on the ts-import repo about it?

Reply:
Looks like this was fixed on Dec 5: https://github.com/radarsu/ts-import/issues/24#issuecomment-1342350870 with version `4.0.0-beta.10`. I won't be able to get back to this PR for a bit, but documenting this